### PR TITLE
Rename more machines (small vs large) and add runtime checks

### PR DESCRIPTION
**Pull request by leonardoalt**

### CODE REVIEW

File: std/machines/binary.asm [hunk 1]
```diff
@@ -32,36 +31,3 @@ machine ByteBinary with
         }
     };
 }
-
-machine Binary(byte_binary: ByteBinary) with
```
Comment (leonardoalt) [inline std/machines/binary.asm:36]: This moved to `large_field::binary`

File: pipeline/tests/powdr_std.rs [hunk 2]
```diff
@@ -107,29 +108,29 @@ fn arith_large_test() {
...
 }
 
 #[test]
 #[ignore = "Too slow"]
-fn memory_test_parallel_accesses() {
-    let f = "std/memory_test_parallel_accesses.asm";
-    regular_test(f, &[]);
+fn memory_large_test_parallel_accesses() {
+    let f = "std/memory_large_test_parallel_accesses.asm";
+    regular_test_without_babybear(f, &[]);
```
Comment (leonardoalt) [inline pipeline/tests/powdr_std.rs:127]: These fail the field size check now. Not sure how they passed before with bb.

File: pipeline/tests/powdr_std.rs [hunk 3]
```diff
@@ -211,34 +212,41 @@ fn write_once_memory_test() {
...
     test_pilcom(make_simple_prepared_pipeline(f));
     test_halo2(make_simple_prepared_pipeline(f));
 }
 
+#[test]
+#[ignore = "Too slow"]
+fn shift_small_test() {
+    let f = "std/shift_small_test.asm";
+    test_plonky3_with_backend_variant::<BabyBearField>(f, vec![], BackendVariant::Composite);
+}
```
Comment (leonardoalt) [inline pipeline/tests/powdr_std.rs:248]: This test existed but wasn't used.

File: std/machines/large_field/binary.asm [hunk 4]
```diff
@@ -0,0 +1,39 @@
+use std::utils::unchanged_until;
+use std::field::modulus;
+use std::check::assert;
+use std::machines::binary::ByteBinary;
+
```
Comment (leonardoalt) [inline std/machines/large_field/binary.asm:5]: Copied from `std::machines::binary::Binary`

File: std/machines/large_field/binary.asm [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+machine Binary(byte_binary: ByteBinary) with
+    latch: latch,
+    operation_id: operation_id,
+    // Allow this machine to be connected via a permutation
+    call_selectors: sel,
+{
+    assert(modulus() > 2**32, || "Binary requires a field that fits any 32-Bit value.");
```
Comment (leonardoalt) [inline std/machines/large_field/binary.asm:12]: + this added

**Comment by leonardoalt** [comment]
> We could introduce a function to std::check that is a bit more descriptive:
> 
> ```
> let require_field_bits = |b, err_f| assert(modulus() > 2**bits, err_f);
> ```

makes sense, will do

**Review by pacheco — APPROVED** [review]

**Pull request merged**